rules/interface_locks.go

```diff
@@ -0,0 +1,172 @@
+package rules
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
+var _ tflint.Rule = new(TerraformLockInterfaceRule)
+
+type TerraformLockInterfaceRule struct {
+	tflint.DefaultRule
+}
+
+func NewTerraformLockInterfaceRule() *TerraformLockInterfaceRule {
+	return new(TerraformLockInterfaceRule)
+}
+
+func (t *TerraformLockInterfaceRule) Name() string {
+	return "tfintlock"
+}
+
+func (t *TerraformLockInterfaceRule) Link() string {
+	return "https://azure.github.io/Azure-Verified-Modules/specs/shared/interfaces/#resource-locks"
+}
+
+func (t *TerraformLockInterfaceRule) Enabled() bool {
+	return false
+}
+
+func (t *TerraformLockInterfaceRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+func (t *TerraformLockInterfaceRule) Check(r tflint.Runner) error {
+	path, err := r.GetModulePath()
+	if err != nil {
+		return err
+	}
+	if !path.IsRoot() {
+		// This rule does not evaluate child modules.
+		return nil
+	}
+	body, err := r.GetModuleContent(&hclext.BodySchema{
+		Blocks: []hclext.BlockSchema{
+			{
+				Type:       "variable",
+				LabelNames: []string{"name"},
+				Body: &hclext.BodySchema{
+					Attributes: []hclext.AttributeSchema{
+						{Name: "type"},
+						{Name: "default"},
+					},
+				},
+			},
+		},
+	}, &tflint.GetModuleContentOption{ExpandMode: tflint.ExpandModeNone})
+	if err != nil {
+		return err
+	}
+
+	for _, variable := range body.Blocks {
+		if variable.Labels[0] != "lock" {
+			continue
+		}
+		// Check if the variable has a type attribute and that it is correct.
+		typeattr, exists := variable.Body.Attributes["type"]
+		if !exists {
+			if err := r.EmitIssue(
+				t,
+				fmt.Sprintf("`%s` variable type not declared", variable.Labels[0]),
+				variable.DefRange,
+			); err != nil {
+				return err
+			}
+			continue
+		}
+		if err := checkLockInterface(typeattr); err != nil {
```

Comment:
Could we rename this `err` to `interfaceErr` so we can rename `err2` in line 82 to `err` to keep all err names aligned?

Reply:
Done